lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -167,16 +165,10 @@ def log_discard(length, found)
     end
 
     def reduce_to_single_packet
-      if @data.length <= 0
-        # Need some data
-        return :STOP
-      end
-
```

Comment:
@ryanatball I deleted this because it doesn't seem reachable. I don't think you can get here without some data in @data.